### PR TITLE
EASY-1945: upgrade dependencies

### PR DESCRIPTION
**Pull request by pfrolke**
Fixes EASY-1945

#### When applied it will
* upgrade parent POM version
* remove self-contained dependency version tags
* fix issues caused by breaking changes of dependency updates

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.solr4files/package.scala [hunk 1]
```diff
@@ -119,8 +119,8 @@ package object solr4files extends DebugEnhancedLogging {
   }
 
   implicit class RichParams(val left: scalatra.Params) extends AnyVal {
-    def asString: String = if (left.isEmpty) "no params at all"
-                           else "params " + left.mkString("[", ",", "]")
+    def asString: String = if (left.size == 0) "no params at all"
+                           else "params " + left.toMap.mkString("[", ",", "]")
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.solr4files/package.scala]: why are these changes necessary? Tried without these and it still builds...

**Review by rvanheest — CHANGES_REQUESTED** [review]
Great to see you also fixed some deprecation warnings. However, I still see some of them regarding

```
AppSpec.scala:66: method getContentStreams in class SolrRequest is deprecated: see corresponding Javadoc for more information.
```

Could you take a look at them?

File: src/main/scala/nl.knaw.dans.easy.solr4files/package.scala (hunk 1 above)
Comment (pfrolke) [reply src/main/scala/nl.knaw.dans.easy.solr4files/package.scala]: This change was necessary when trying to a newer milestone version of scalatra. But since the final parent POM only contains the latest stable version, these changes are indeed no longer needed.

**Comment by pfrolke** [comment]
@rvanheest I had to change two tests in SolrErrorHandlingSpec quite a bit in order to get it to work with solrj-solrj version > 8.0.0. Could you confirm that it is still working as expected?

File: src/test/scala/nl.knaw.dans.easy.solr4files/SolrErrorHandlingSpec.scala [hunk 2]
```diff
@@ -68,8 +68,8 @@ class SolrErrorHandlingSpec extends TestSupportFixture
 
   "delete" should "return the exception bubbling up from solrClient.deleteByQuery" in {
     delete("/fileindex/?q=:") {
-      body shouldBe "Error from server at mockedHost: mocked parser"
-      status shouldBe SC_BAD_REQUEST
+      body shouldBe "solr delete [:] failed with Error from server at mockedHost: mocked parser"
+      status shouldBe SC_INTERNAL_SERVER_ERROR
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.solr4files/SolrErrorHandlingSpec.scala]: why did these status codes change? Perhaps an exception that has changed?

Comment (pfrolke) [reply src/test/scala/nl.knaw.dans.easy.solr4files/SolrErrorHandlingSpec.scala]: I think so, the tests started failing when upgrading `solrj-solrj` from version `7.7.2` to `8.0.0` and higher. However, I can't find information about this change in the [changelogs](https://lucene.apache.org/solr/guide/8_0/major-changes-in-solr-8.html). Do you know if this change would cause other modules interacting with this module to fail?

Comment (rvanheest) [reply src/test/scala/nl.knaw.dans.easy.solr4files/SolrErrorHandlingSpec.scala]: Turns out, there were a number of deprecation warnings in the code. After fixing those, the tests behaved as specified before.

**Review by rvanheest — APPROVED** [review]
